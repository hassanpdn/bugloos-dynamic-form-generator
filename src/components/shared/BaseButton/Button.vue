<template>
      <button class="
            p-2.5
            rounded-lg
            hover:shadow-md 
            hover:shadow-gray-100 
            transition-colors 
            transition-shadows 
            duration-500
            ease-linear"
            :class="`${expanded ? 'w-full' : ''} ${disabled ? 'bg-gray-400' : ''} hover:bg-${bgColor}-50 bg-${bgColor}-300 text-${textColor}`"
            :disabled="disabled"
      >
            {{ text }}
      </button>
</template>

<script lang="ts">
      import { defineComponent, PropType } from 'vue';
      export default defineComponent({
            name: 'button-component',
            props: {
                  expanded: {
                        type: Boolean as PropType<boolean>,
                        default: false
                  },
                  textColor: {
                        type: String as PropType<string>,
                        default: 'black'
                  },
                  bgColor: {
                        type: String as PropType<string>,
                        default: "bg-blue-500"
                  },
                  text: {
                        type: String as PropType<string>,
                        default: "button"
                  },
                  disabled: {
                        type: Boolean as PropType<boolean>,
                        default: false
                  }
            }
      })
</script>

<style>
</style>