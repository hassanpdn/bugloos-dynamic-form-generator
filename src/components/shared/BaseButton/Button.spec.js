import { mount } from '@vue/test-utils'
import Button from '@/components/shared/BaseButton/Button.vue';
import { describe, it, expect } from "vitest";

describe('Button', () => {
      const wrapper = mount(Button)
      it('renders a vue instance', () => {
            expect(wrapper.html()).toMatchSnapshot()
      })

      it('renders a button', () => {
            expect(wrapper.find('button').exists()).toBe(true)
      })

      it('should call click handler', async () => {
            await wrapper.find('button').trigger('click')
            expect(wrapper.emitted().click).toBeTruthy()
      })

      it('checks the prop to be passed', () => {
            const wrapper = mount(Button, {
                  propsData: {
                        expanded: true,
                        bgColor: 'red'
                  }
            })
            expect(wrapper.classes()).toContain('w-full');
            expect(wrapper.classes()).toContain('bg-red-300');
      })

})