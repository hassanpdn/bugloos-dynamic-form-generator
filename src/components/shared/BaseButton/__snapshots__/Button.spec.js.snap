// Vitest Snapshot v1

exports[`Button > renders a vue instance 1`] = `"<button class=\\"p-2.5 rounded-lg hover:shadow-md hover:shadow-gray-100 transition-colors transition-shadows duration-500 ease-linear hover:bg-bg-blue-500-50 bg-bg-blue-500-300 text-black\\">button</button>"`;
